dhis-2/dhis-web/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/GistQueryControllerTest.java

```diff
@@ -0,0 +1,313 @@
+    {
+        String url = "/users/{uid}/userGroups/gist?fields=name,users::not-member({uid})";
+
+        // not-member(id) with a user that is a member
+        JsonObject gist = GET( url, getSuperuserUid(), getSuperuserUid() ).content();
+        assertFalse( gist.getArray( "userGroups" ).getObject( 0 ).getBoolean( "users" ).booleanValue() );
+
+        // not-member(id) with a user that is not a member
+        gist = GET( url, getSuperuserUid(), "non-existing-user-uid" ).content();
+        assertTrue( gist.getArray( "userGroups" ).getObject( 0 ).getBoolean( "users" ).booleanValue() );
+    }
+
+    @Test
+    public void testGistPropertyList_PagerWithTotal_ResultBased()
+    {
+        JsonObject gist = GET( "/users/{uid}/userGroups/gist?fields=name,users&total=true",
+            getSuperuserUid() ).content();
+
+        assertHasPager( gist, 1, 50, 1 );
+    }
+
+    @Test
+    public void testGistPropertyList_PagerWithTotal_CountQuery()
+    {
+        // create some members we can count a total for
+        createDataSetsForOrganisationUnit( 10, orgUnitId, "extra" );
+
+        String url = "/organisationUnits/{id}/dataSets/gist?total=true&pageSize=3&order=name&filter=name:startsWith:extra";
+        JsonObject gist = GET( url, orgUnitId ).content();
+        assertHasPager( gist, 1, 3, 10 );
+
+        // now page 2
+        gist = GET( url + "&page=2", orgUnitId ).content();
+        assertHasPager( gist, 2, 3, 10 );
+        assertEquals(
+            "/organisationUnits/{id}/dataSets/gist?total=true&pageSize=3&order=name&filter=name%3AstartsWith%3Aextra&page=1"
+                .replace( "{id}", orgUnitId ),
+            gist.getObject( "pager" ).getString( "prevPage" ).string() );
+        assertEquals(
+            "/organisationUnits/{id}/dataSets/gist?total=true&pageSize=3&order=name&filter=name%3AstartsWith%3Aextra&page=3"
+                .replace( "{id}", orgUnitId ),
+            gist.getObject( "pager" ).getString( "nextPage" ).string() );
+        JsonArray dataSets = gist.getArray( "dataSets" );
+        assertEquals( "extra3", dataSets.getObject( 0 ).getString( "name" ).string() );
+        assertEquals( "extra4", dataSets.getObject( 1 ).getString( "name" ).string() );
+        assertEquals( "extra5", dataSets.getObject( 2 ).getString( "name" ).string() );
+    }
+
+    @Test
+    public void testGistPropertyList_FilterByIn()
+    {
+        // create some items we can filter
+        createDataSetsForOrganisationUnit( 10, orgUnitId, "plus" );
+
+        String url = "/organisationUnits/{id}/dataSets/gist?filter=name:in:[plus3,plus5]&fields=name&order=name";
+        JsonObject gist = GET( url, orgUnitId ).content();
+
+        assertEquals( asList( "plus3", "plus5" ), gist.getArray( "dataSets" ).stringValues() );
+    }
+
+    @Test
+    public void testGistPropertyList_EndpointsWithAbsoluteURLs()
+    {
+        JsonObject groups = GET( "/users/{uid}/userGroups/gist?fields=name,users&absolute=true",
```

Comment:
Could we perhaps change the `absolute` query param to `absoluteUrls` to make it more descriptive and avoid future naming conflicts?

Reply:
changed.